components/script/script_thread.rs

```diff
@@ -1761,6 +1758,13 @@ impl ScriptThread {
         ROUTER.route_ipc_receiver_to_mpsc_sender(ipc_timer_event_port,
                                                  self.timer_event_chan.clone());
 
+        // If this is an about:blank load, it must share the creator's origin.
```

Comment:
Shouldn't this logic be: 
```
   let origin = if url != "about:blank" {
      url.origin()
   } else if parent exists {
      parent's origin
   } else if creator exists {
      creator's origin
   } else {
      new opaque origin
   };
```
(Not sure about whether to use `final_url` or `incomplete.url` here, there is probably some horrorshow about pages redirecting to `about:blank` we have to worry about.)